src/main/java/org/opensearch/flowframework/workflow/ToolStep.java

```diff
@@ -114,6 +115,13 @@ private Map<String, String> getToolsParametersMap(
     ) {
         @SuppressWarnings("unchecked")
         Map<String, String> parametersMap = (Map<String, String>) parameters;
+
+        Optional<String> previousNodeConnector = previousNodeInputs.entrySet()
+            .stream()
+            .filter(e -> CONNECTOR_ID.equals(e.getValue()))
+            .map(Map.Entry::getKey)
+            .findFirst();
+
```

Comment:
Can we create a common method for L119-127 and just filter based on the argument passed as `CONNECTOR_ID, MODEL_ID, AGENT_ID`?

Reply:
Updated.